src/test/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusServiceSpec.scala

```diff
@@ -350,13 +380,261 @@ class ConsignmentStatusServiceSpec extends AnyFlatSpec with MockitoSugar with Re
+      Seq(
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now())),
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now())),
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Failed", Timestamp.from(Instant.now()))
+      )
+    )
+    when(
+      consignmentStatusRepositoryMock.updateConsignmentStatus(
+        any[UUID],
+        any[String],
+        statusValueCaptor.capture(),
+        any[Timestamp]
+      )
+    ).thenReturn(consignmentStatusMockRepoResponse)
+
+    when(
+      fileStatusRepositoryMock.getFileStatus(
+        expectedConsignmentId,
+        expectedStatusType
+      )
+    ).thenReturn(fileStatusMockRepoResponse)
+
+    val updateConsignmentStatusInput = ConsignmentStatusInput(expectedConsignmentId, expectedStatusType, None)
+
+    consignmentService.updateConsignmentStatus(updateConsignmentStatusInput).futureValue
+
+    statusValueCaptor.getValue should equal(expectedStatusValue)
+  }
+
+  "updateConsignmentStatus" should s"(given an Upload statusType) generate a statusValue based on the file status, even when a statusValue was passed in" in {
+    val fixedUUIDSource = new FixedUUIDSource()
+    val expectedConsignmentId = fixedUUIDSource.uuid
+    val expectedStatusType = "Upload"
+    val passedInStatusValue = "InProgress"
+    val expectedStatusValue = "Completed"
+    val statusValueCaptor: ArgumentCaptor[String] = ArgumentCaptor.forClass(classOf[String])
+
+    val consignmentStatusMockRepoResponse: Future[Int] = Future.successful(1)
+    val fileStatusMockRepoResponse: Future[Seq[FilestatusRow]] = Future.successful(
+      Seq(
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now())),
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now())),
+        FilestatusRow(UUID.randomUUID(), UUID.randomUUID(), "Upload", "Success", Timestamp.from(Instant.now()))
+      )
+    )
+    when(
+      consignmentStatusRepositoryMock.updateConsignmentStatus(
+        any[UUID],
+        any[String],
+        statusValueCaptor.capture(),
+        any[Timestamp]
+      )
+    ).thenReturn(consignmentStatusMockRepoResponse)
+
+    when(
+      fileStatusRepositoryMock.getFileStatus(
+        any[UUID],
+        any[String],
+        any[Option[Set[UUID]]]
+      )
+    ).thenReturn(fileStatusMockRepoResponse)
+
+    val updateConsignmentStatusInput =
+      ConsignmentStatusInput(expectedConsignmentId, expectedStatusType, Some(passedInStatusValue))
+
+    consignmentService.updateConsignmentStatus(updateConsignmentStatusInput).futureValue
+
+    statusValueCaptor.getValue should equal(expectedStatusValue)
+  }
+
+  "updateConsignmentStatus" should s"throw an InputDataException if an Upload statusType was provided, a statusValue wasn't provided " +
```

Comment:
Maybe reword the test title slightly to: `... for an Upload statusType with no statusValue where no consignment files have an Upload status`